File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int ans = 0;
        int profit = 0;
        int buy = prices[0];
        for(int i=0; i<prices.size(); i++)
        {
            if(prices[i] < buy)
            {
                buy = prices[i];
            }
            else
            {
                profit = prices[i]-buy;
                 ans = max(profit,ans);
            }
        }
        return ans;
    }
};

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        buy = prices[0]
        profit = 0
        ans = 0
        for i in range(len(prices)):
            if prices[i] < buy:
                buy = prices[i]
            else:
                profit = prices[i] - buy
                ans = max(ans,profit)
        return ans
        